MdsSupportingClasses/Collivery.php

```diff
@@ -1121,36 +1053,49 @@ public function addCollivery(array $data)
             $this->setError('invalid_data', 'Invalid Contact ID for: contact_to.');
         }
 
-        if (!isset($data['collivery_type'])) {
-            $this->setError('missing_data', 'collivery_type not set.');
-        } elseif (!isset($parcel_types[$data['collivery_type']])) {
-            $this->setError('invalid_data', 'Invalid collivery_type.');
-        }
-
         if (!isset($data['service'])) {
             $this->setError('missing_data', 'service not set.');
         } elseif (!isset($services[$data['service']])) {
             $this->setError('invalid_data', 'Invalid service.');
         }
 
         if (!$this->hasErrors()) {
+
+            $newObject = [];
+
+            $newObject = array(
+                "service" => $data["service"],
+                "parcels" => $data["parcels"],
+                "collection_address" => $data["collivery_from"],
+                "collection_contact" => $data["contact_from"],
+                "delivery_address" => $data["collivery_to"],
+                "delivery_contact" => $data["contact_to"],
+                "collection_time" => $data["collection_time"],
+                "exclude_weekend" => true,
+                "risk_cover" => $data["cover"],
+                "special_instructions" => $data["instructions"],
+                "reference" => $data["cust_ref"]
+            );
+
             try {
-                $result = $this->client()->add_collivery($data, $this->token);
+                $result = $this->consumeAPI("https://api.collivery.co.za/v3/waybill", $newObject, 'POST');
+                new SoapConnectionException("New Object => ".json_encode($newObject));
```

Comment:
If the `SoapConnectionException` is used solely for logging, please rather use the `MdsLogger` class directly

Reply:
Apologies, I forgot to remove that during development. 
I've updated to remove any unnecessary logging.